.devcontainer/blog/devcontainer.json

```diff
@@ -0,0 +1,43 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.238.1/containers/docker-existing-dockerfile
+{
+    "name": "Astro Examples",
+    "build": {
+      "dockerfile": "../examples.Dockerfile"
+    },
+
+    "workspaceFolder": "/workspaces/astro/examples/blog",
+
+    "portsAttributes": {
+      "3000": {
+        "label": "Application",
+        "onAutoForward": "openPreview"
+      }
+    },
+  
+    "forwardPorts": [3000],
+  
+    "postCreateCommand": "pnpm install && cd /workspaces/astro && pnpm run build",
+  
+    "waitFor": "postCreateCommand",
+  
+    "postAttachCommand": {
+      "Server": "pnpm start --host"
+    },
+  
+    "customizations": {
+      "codespaces": {
+        "openFiles": [
+          "src/pages/index.astro"
+        ]
+      },
+      "vscode": {
+        "extensions": [
+          "astro-build.astro-vscode",
+          "esbenp.prettier-vscode",
+          "dbaeumer.vscode-eslint"
```

Comment:
Same comment here and for every other devcontainer.json which is based upon the javascript-node image